test/testdrive/apd-sum.td

```diff
@@ -0,0 +1,353 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Note that this test uses an append-only source, so doesn't fully express the
+# semantics of summing APD values because values are non-retractable. # A future
+# iteration of this test will use a more robust source and improve the tests # to
+# more accurately reflect the operation's semantics.
+
+$ file-append path=values.csv
+1.2
+2.3
+3.4
+
+> CREATE MATERIALIZED SOURCE apd_csv
+  FROM FILE '${testdrive.temp-dir}/values.csv' WITH (tail = true)
+  FORMAT CSV WITH 1 COLUMNS
+
+> CREATE VIEW apd_values AS
+    SELECT column1::apd AS a
+    FROM apd_csv;
+
+> CREATE MATERIALIZED VIEW apd_values_sum AS
+    SELECT sum(a) AS sum_a FROM apd_values;
+
+> SELECT sum_a::text AS sum_a FROM apd_values_sum;
+sum_a
+----
+6.9
+
+$ file-append path=values.csv
+0.0
+
+> SELECT sum_a::text AS sum_a FROM apd_values_sum;
+sum_a
+----
+6.9
+
+$ file-append path=values.csv
+-1.2
+-2.3
+-3.4
+-1.2
+-2.3
+-3.4
+
+> SELECT sum_a::text AS sum_a FROM apd_values_sum;
+sum_a
+----
+-6.9
+
+$ file-append path=values.csv
+1.2
+2.3
+3.4
+
+> SELECT sum_a::text AS sum_a FROM apd_values_sum;
+sum_a
+----
+0
+
+# sum operation preserves commutativity, even when it appears lost from the
+# datum's perspective.
+
+$ file-append path=values.csv
+1e38
+
+> SELECT sum_a::text AS sum_a FROM apd_values_sum;
+sum_a
+----
+100000000000000000000000000000000000000
+
+$ file-append path=values.csv
+9e38
+
+# When you "fill" >38 digits of precision in the aggregator, signal
+# pseudo-overflow with infinity.
+> SELECT sum_a::text AS sum_a FROM apd_values_sum;
+sum_a
+----
+Infinity
```

Comment:
So we have both a case where SUM returns "Infinity" and a case where it throws a "numeric field overflow". This needs to be documented as the "infinity" part is not intuitive. You can not seem to put the result from such a SUM() back into a DECIMAL/APD column:

```
materialize=> select 'Infinity'::apd;
ERROR:  invalid input syntax for type apd: "Infinity"
```

Reply:
Yeah, this will definitely be a large part of the documentation. Will start doing that now with inline comments in the testdrive file.

The reasoning behind the two different behaviors is where the actual "overflow" occurs:
- In the case of getting `numeric field overflow`, the aggregated value is getting rescaled, e.g. cast to `apd(39,3)`, and the rescaled value is invalid because it would cause the loss of significant digits > 1. This is a failure of a unary function getting applied to a scalar value, i.e. the aggregated result.
- In the case of getting an "Infinity" result, the aggregated value is >= `1e39`, i.e. it's too large to fit in the APD datum. This behavior is necessary to ensure we don't lose associativity/commutativity and only occurs in aggregation contexts. So this is the unexpected behavior from other type in Materialize and will definitely require a lot of communication.